src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -67,22 +67,37 @@
       final Collection<Unit> transports) {
 
     final List<Unit> canBeTransported = sortByTransportCostDescending(units);
-    final List<Unit> canTransport = sortByTransportCapacityAscendingThenMovesDescending(transports);
+    final List<Unit> canTransport = sortByTransportCapacityDescendingThenMovesDescending(transports);
 
     // Add max units to each transport
     final Map<Unit, Unit> mapping = new HashMap<>();
+    Optional<Unit> finalTransport = Optional.empty();
     for (final Unit transport : canTransport) {
-      int capacity = TransportTracker.getAvailableCapacity(transport);
-      for (final Iterator<Unit> it = canBeTransported.iterator(); it.hasNext();) {
-        final Unit unit = it.next();
-        final int cost = UnitAttachment.get((unit).getType()).getTransportCost();
-        if (capacity >= cost) {
-          capacity -= cost;
-          mapping.put(unit, transport);
-          it.remove();
+
+      // Check if remaining units can all be loaded into 1 transport
+      final int capacity = TransportTracker.getAvailableCapacity(transport);
+      final int remainingCost = getTransportCost(canBeTransported);
+      if (remainingCost <= capacity) {
+        if (!finalTransport.isPresent() || capacity < TransportTracker.getAvailableCapacity(finalTransport.get())) {
+          finalTransport = Optional.of(transport);
```

Comment:
Loop needs to be sorted with the largest transport first since we are trying to use min number of transports. The exception to this is the last transport being loaded. Example is say we have a small transport (holds 1) and a big transport (holds 2) and we have 1 unit left, we want to load the unit in the small transport as this wastes the minimum amount of space.